toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/StartingPointDetailsScreen.kt

```diff
@@ -17,72 +17,288 @@
-import com.arcgismaps.geometry.Polyline
-import com.arcgismaps.toolkit.utilitynetworks.ui.material3.Slider
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
 import com.arcgismaps.toolkit.utilitynetworks.R
 import com.arcgismaps.toolkit.utilitynetworks.StartingPoint
+import com.arcgismaps.toolkit.utilitynetworks.internal.util.Title
+import com.arcgismaps.toolkit.utilitynetworks.ui.material3.Slider
+import com.arcgismaps.utilitynetworks.UtilityNetworkSourceType
+import com.arcgismaps.utilitynetworks.UtilityTerminal
 
-
+/**
+ * A composable screen that shows the details of a starting point for a trace.
+ *
+ * @since 200.6.0
+ */
 @Composable
-internal fun StartingPointDetailsScreen(startingPoint: StartingPoint?,
+internal fun StartingPointDetailsScreen(startingPoint: StartingPoint,
+                                        onZoomTo: () -> Unit,
+                                        onDelete: () -> Unit,
                                         onFractionChanged: (StartingPoint, Float) -> Unit,
+                                        onTerminalSelected: (UtilityTerminal) -> Unit,
                                         onBackPressed: () -> Unit) {
     Surface(modifier = Modifier.fillMaxSize()) {
         Column(
             modifier = Modifier.fillMaxWidth(),
             horizontalAlignment = Alignment.Start,
             verticalArrangement = Arrangement.Top) {
-            Row(
-                modifier = Modifier.clickable { onBackPressed() },
-                horizontalArrangement = Arrangement.SpaceBetween,
-                verticalAlignment = Alignment.CenterVertically
-            ) {
-            Image(
-                imageVector = Icons.AutoMirrored.Default.ArrowBack,
-                contentDescription = "back"
+
+            BackButton(onBackPressed)
+
+            Spacer(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(10.dp)
             )
-            ReadOnlyTextField(
-                text = stringResource(id = R.string.starting_points))
-            }
-        }
 
-        startingPoint?.let { point ->
-            val geometry = point.feature.geometry
+            Title(startingPoint.name, onZoomTo, onDelete)
 
-            if (geometry != null && geometry is Polyline ) {
-                FractionAlongEdgeSlider(point, onFractionChanged)
+            Spacer(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(10.dp)
+            )
+
+            LazyColumn(modifier = Modifier
+                .padding(vertical = 3.dp),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                if (startingPoint.utilityElement.networkSource.sourceType == UtilityNetworkSourceType.Edge) {
+                    item {
+                        FractionAlongEdgeSlider(startingPoint, onFractionChanged)
```

Comment:
Then you can replace the call as such: 
```suggestion
                        FractionAlongEdgeSlider(
                            fractionAlongEdge = startingPoint.utilityElement.fractionAlongEdge.toFloat(),
                            onFractionChanged = { newValue ->
                                onFractionChanged(startingPoint, newValue)
                            }
                        )
```

Reply:
I don't see the advantage in doing this. I like this call to be simple and all the functionality contained in the function.